### PR TITLE
feat(rds): support more AWS RDS DB Instance engines in encryption check

### PR DESCRIPTION
**Pull request by madereddy**
### Context

Add additional RDS transport level encryption logic for supported RDS versions:

> For SQL Server, PostgreSQL and Aurora PostgreSQL databases, if the rds.force_ssl parameter value is set to 0, the Transport Encryption feature is not enabled. For MySQL, Aurora MySQL and MariaDB databases, if the require_secure_transport parameter value is set to OFF, the Transport Encryption feature is not enabled.

### Description

Added additional checks for ```MySQL, MariaDB, Aurora PostgreSQL, and Aurora MySQL``` DB instances.

Corrected Microsoft SQL Server checks as well to include all the engine versions available.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py [hunk 1]
```diff
@@ -75,6 +75,45 @@ def test_rds_aurora_instance(self):
...
+            ):
+                # Test Check
+                from prowler.providers.aws.services.rds.rds_instance_transport_encrypted.rds_instance_transport_encrypted import (
+                    rds_instance_transport_encrypted,
+                )
+
+                check = rds_instance_transport_encrypted()
+                result = check.execute()
+
+                assert len(result) == 1
```
Comment (jfagoagas) [inline tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py:93]: Here you need to assert all the fields in the result object, could you please do that? Thanks!

File: tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py [hunk 2]
```diff
@@ -156,7 +195,131 @@ def test_rds_instance_no_ssl(self):
...
+                assert result[0].status == "FAIL"
+                assert search(
```
Comment (jfagoagas) [inline tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py]: Could you please assert the whole line in the `status_extended` without using the `search` function?

Comment (jfagoagas) [reply tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py]: The same for the next test.

**Review by jfagoagas — CHANGES_REQUESTED** [review]
That's a great addition @madereddy 👏 

I left you some comments in the test cases to improve them, thanks again fro your contribution 🚀

**Comment by MrCloudSec** [comment]
I have added the context for these changes in the description:

> For SQL Server, PostgreSQL and Aurora PostgreSQL databases, if the rds.force_ssl parameter value is set to 0, the Transport Encryption feature is not enabled. For MySQL, Aurora MySQL and MariaDB databases, if the require_secure_transport parameter value is set to OFF, the Transport Encryption feature is not enabled.

**Comment by jfagoagas** [comment]
> I have added the context for these changes in the description:
> 
> > For SQL Server, PostgreSQL and Aurora PostgreSQL databases, if the rds.force_ssl parameter value is set to 0, the Transport Encryption feature is not enabled. For MySQL, Aurora MySQL and MariaDB databases, if the require_secure_transport parameter value is set to OFF, the Transport Encryption feature is not enabled.

@sergargar @madereddy could you please add this to the check's metadata? thanks!

**Comment by madereddy** [comment]
@jfagoagas The checks metadata will be updated along with the integration tests with the next commit. 

Thank you @sergargar for the updates.

File: tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py (hunk 1 above)
Comment (madereddy) [reply tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py:93]: Updated in new commit

File: tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py (hunk 2 above)
Comment (madereddy) [reply tests/providers/aws/services/rds/rds_instance_transport_encrypted/rds_instance_transport_encrypted_test.py]: Updated in new commit

**Comment by jfagoagas** [comment]
@madereddy I think you included changes from other commits in your last commit, please take a look a it. Thanks!

**Comment by madereddy** [comment]
@jfagoagas Looks like I pulled the most recent commits and did not back it out successfully. I will let you know when it has been updated correctly.

**Pull request closed without merging**